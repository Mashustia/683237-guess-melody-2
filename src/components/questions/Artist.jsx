import React from 'react';
import PropTypes from 'prop-types';

export const Artist = (props) => {
  const {name, picture, onChoose} = props;

  return (
    <div className='artist'>
      <input className='artist__input visually-hidden' type='radio' name='answer' value='artist-2' id='answer-2'/>
      <label className='artist__name' htmlFor='answer-2' onClick={onChoose(name)}>
        <img className='artist__picture' src={picture} alt='Пелагея'/>
        {name}
      </label>
    </div>
  );
};

Artist.propTypes = {
  name: PropTypes.string,
  picture: PropTypes.string,
  onChoose: PropTypes.func.isRequired,
};

export default Artist;
