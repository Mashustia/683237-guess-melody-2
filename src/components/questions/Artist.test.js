import React from 'react';
import renderer from 'react-test-renderer';
import Artist from './Artist';

it(`Artist component renders correctly`, () => {
  const component = renderer
    .create(<Artist name={``} picture={``} onChoose={jest.fn()} />)
    .toJSON();
  expect(component).toMatchSnapshot();
});
