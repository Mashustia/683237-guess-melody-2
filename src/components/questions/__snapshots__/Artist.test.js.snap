// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Artist component renders correctly 1`] = `
<div
  className="artist"
>
  <input
    className="artist__input visually-hidden"
    id="answer-2"
    name="answer"
    type="radio"
    value="artist-2"
  />
  <label
    className="artist__name"
    htmlFor="answer-2"
  >
    <img
      alt="Пелагея"
      className="artist__picture"
      src=""
    />
    
  </label>
</div>
`;
